lld/COFF/Symbols.cpp

```diff
@@ -126,9 +127,14 @@ DefinedImportThunk::DefinedImportThunk(COFFLinkerContext &ctx, StringRef name,
 
 Defined *Undefined::getWeakAlias() {
   // A weak alias may be a weak alias to another symbol, so check recursively.
-  for (Symbol *a = weakAlias; a; a = cast<Undefined>(a)->weakAlias)
+  SmallSet<Symbol *, 4> weakChain;
```

Comment:
DenseMap, which I believe has the least code size increase.

SmallPtrSet would be better than SmallSet but its small container case compiles to additional code than DenseMap.

Reply:
So DenseMap is better here, than using DenseSet?

Reply:
Looking at `DenseSet`, it's a thin wrapper around `DenseMap`, so it seems fine to me. I will change it, please let me know if I should revert.